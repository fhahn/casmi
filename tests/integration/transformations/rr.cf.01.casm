// transform, cmdline "--frr --fcf"

CASM lr

init foo

rule foo =
{
    let x = 10 in
    {
        print x
    }

    let x = 10 in
    {
        print x
    }

    let x = 10 in
    {
        print x
    }

    program(self) := undef
}


// step0: lr
// rule foo =
// {
//     __l0 := 10 
//     print __l0
//
//     __l1 := 10 
//     print __l1
//
//     __l2 := 10 
//     print __l2
//     program(self) := undef
// }

// step1: lr.cf
// rule foo =
// {
//     __c0 =: 10
//
//     __l0 := 10 
//     print __c0
//
//     __l1 := 10 
//     print __c0
//
//     __l2 := 10 
//     print __c0
//
//     program(self) := undef
// }

// step2: lr.cf.cp
// rule foo =
// {
//     __c0 =: 10
//
//     __l0 := __c0 
//     print __l0
//
//     __l1 := __c0 
//     print __l1
//
//     __l2 := __c0 
//     print __l2
//
//     program(self) := undef
// }

// step3: lr.cf.cp.cp
// rule foo =
// {
//     __c0 =: 10
//
//     print __c0
//
//     print __c0
//
//     print __c0
//
//     program(self) := undef
// }




