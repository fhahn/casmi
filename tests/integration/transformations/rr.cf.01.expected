foo=STATEMENTS(LET CONST(ident c0__,num 10),STATEMENTS(LET CONST(ident c1__,SELF),STATEMENTS(LET CONST(ident c2__,UNDEF),PARBLOCK(STATEMENTS(STATEMENTS(SKIP,PARBLOCK(PARBLOCK(STATEMENTS(PRINT(LIST(ident c0__,null),null),null),null),null)),STATEMENTS(STATEMENTS(SKIP,PARBLOCK(PARBLOCK(STATEMENTS(PRINT(LIST(ident c0__,null),null),null),null),null)),STATEMENTS(STATEMENTS(SKIP,PARBLOCK(PARBLOCK(STATEMENTS(PRINT(LIST(ident c0__,null),null),null),null),null)),STATEMENTS(UPDATE(LOOKUP(ident program,LIST(ident c1__,null)),ident c2__),null)))),null))))